### PR TITLE
test/extended/cli/mustgather: Separate gather_audit_logs test

### PR DESCRIPTION
**Pull request by wking**
openshift/must-gather#143 is removing these from the default gather, because they're mostly useful for internal debugging, less useful in end-user bug reports, and can run to hundreds of megabytes.  But we still want to ensure that they work as expected when they are explicitly requested.  This commit pulls the audit-log checks out of the test-case for the generic invocation.  And it adds a new test case with those checks after an explict `gather_audit_logs` request.

CC @deads2k, @sanchezl, @sferich888, @soltysh

[1]: https://github.com/openshift/must-gather/pull/143

### CODE REVIEW

**Comment by deads2k** [comment]
/approve
/lgtm
/hold

holding so you can make it compile.  release and retag after.

**Comment by wking** [comment]
The three e2e failures were:

* GCP upgrade: [`timed out waiting for service "service-test" to have a load balancer`][1].
* [GCP][2] on the new test:
    ```
    fail [github.com/openshift/origin/test/extended/cli/mustgather.go:181]: Unexpected error:
        <*errors.errorString | 0xc0000da050>: {s: "EOF"}
        EOF
    occurred
    ```
    And similarly for [AWS FIPS][3].

We probably don't want to land this until have those sorted.

[1]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24680/pull-ci-openshift-origin-master-e2e-gcp-upgrade/2789
[2]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24680/pull-ci-openshift-origin-master-e2e-gcp/6273
[3]: https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/24680/pull-ci-openshift-origin-master-e2e-aws-fips/1720

**Comment by wking** [comment]
Rebased around the generated-file conflict (from #24682?) with 45a05a1006 -> 769c5f33a1.

I still have no idea what's going on with [the EOF failures][1], maybe someone who understands must-gather can weigh in?

[1]: https://github.com/openshift/origin/pull/24680#issuecomment-598520064

**Comment by soltysh** [comment]
> I still have no idea what's going on with [the EOF failures](https://github.com/openshift/origin/pull/24680#issuecomment-598520064), maybe someone who understands must-gather can weigh in?

It's very weird, I've just tried with the artifacts gathered after the failure and they work fine with similar gzip code. That EOF suggest the file is empty/corrupted/being written to? Can you add a nasty `time.Sleep(10*time.Second)`, honestly I wonder if this https://github.com/openshift/origin/blob/master/test/extended/cli/mustgather.go#L59 one should not be after the call to must-gather, can you give it a try?

**Comment by wking** [comment]
Rebased around 705b5e6987 (#24719) with 769c5f33a1 -> e86b60e2ca.

@soltysh, where exactly did you want that new `Sleep`?

File: test/extended/cli/mustgather.go [hunk 1]
```diff
@@ -122,9 +85,85 @@ var _ = g.Describe("[sig-cli] oc adm must-gather", func() {
...
+		o.Expect(err).ToNot(o.HaveOccurred())
+		defer os.RemoveAll(tempDir)
+
+		args := []string{
+			"--dest-dir", tempDir,
+			"--",
+			"/usr/bin/gather_audit_logs",
+		}
+
+		o.Expect(oc.Run("adm", "must-gather").Args(args...).Execute()).To(o.Succeed())
```
Comment (soltysh) [inline test/extended/cli/mustgather.go:151]: > @soltysh, where exactly did you want that new Sleep?

@wking I was thinking after this call.

**Comment by wking** [comment]
Most recent [GCP failure][1]:

```
[sig-cli] oc adm must-gather runs successfully for audit logs [Suite:openshift/conformance/parallel]
--
fail [github.com/openshift/origin/test/extended/cli/mustgather.go:193]: Unexpected error:
    <*errors.errorString | 0xc0000c6060>: {s: "EOF"}
    EOF
occurred
```

[1]: https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/24680/pull-ci-openshift-origin-master-e2e-gcp/7995

File: test/extended/cli/mustgather.go (hunk 1 above)
Comment (wking) [reply test/extended/cli/mustgather.go:162]: Rebased onto master and added a sleep after this call with e86b60e2ca -> 9282113df1.

**Comment by sferich888** [comment]
@deads2k @soltysh do we have some major flakey tests here? It seems like this has been broken for some 3+ days now.

**Comment by soltysh** [comment]
> @deads2k @soltysh do we have some major flakey tests here? It seems like this has been broken for some 3+ days now.

@sferich888 there are these days :disappointed: and it looks like quay is again suffering, also it's code freeze this week, so it's usually like that :(

**Comment by wking** [comment]
Also, there's no bug link, so even if we do go green we will not merge before 4.5 forks off from master.

**Comment by wking** [comment]
Rebased around #25039 with 9282113df1 -> 6dcbbffc3e.

**Comment by wking** [comment]
[Still getting the EOF][1], even with the additional `Sleep`.

[1]: https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/24680/pull-ci-openshift-origin-master-e2e-gcp/8163

**Comment by sferich888** [comment]
> [Still getting the EOF](https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/24680/pull-ci-openshift-origin-master-e2e-gcp/8163), even with the additional `Sleep`.

Is that the same issue as captured by: https://bugzilla.redhat.com/show_bug.cgi?id=1824007

Should we have @sanchezl and @deads2k help us look into this.

**Comment by soltysh** [comment]
It's actually being tracked here https://bugzilla.redhat.com/show_bug.cgi?id=1829243 @ingvagabund will be looking at it more this week.

**Comment by ingvagabund** [comment]
Wrt. https://bugzilla.redhat.com/show_bug.cgi?id=1829243, `oc adm must-gather` (through invoking `oc adm node-logs`) is just pulling audit files from each master node. In case the audit file is corrupted (e.g. lines missing a prefix), `[cli] oc adm must-gather [Top Level] [cli] oc adm must-gather runs successfully` fails.

**Comment by ingvagabund** [comment]
Checking all the runs in `e2e-gcp` and `e2e-aws-fips` where a cluster was properly provisioned `[sig-cli] oc adm must-gather runs successfully for audit logs` is always failing with:

```
fail [github.com/openshift/origin/test/extended/cli/mustgather.go:199]: Unexpected error:
    <*errors.errorString | 0xc0000ce060>: {s: "EOF"}
    EOF
occurred
```

**Comment by soltysh** [comment]
> Checking all the runs in `e2e-gcp` and `e2e-aws-fips` where a cluster was properly provisioned `[sig-cli] oc adm must-gather runs successfully for audit logs` is always failing with:

@ingvagabund can you try setting up gcp cluster and verify if this works for you, maybe we need to extended timeouts, although `unexpected EOF` from my experience usually shows some networking issues. Maybe we can make the test more tolerant for these?

**Comment by soltysh** [comment]
I've managed to nail down the problems with this PR, updated version for review is in https://github.com/openshift/origin/pull/25283
/close

**Pull request closed without merging**